xtype/type.go

```diff
@@ -177,6 +193,39 @@ func FindField(name string, ignoreCase bool, source *Type, additionalFieldSource
 	}
 }
 
+// protobufSubtagOrder specifies which protobuf tag sub-tags should be considered and in which order.
+var protobufSubtagOrder = [2]string{"json=", "name="}
+
+// parseTagName parses out the field name from a struct tag.
+func parseTagName(matchTag, tag string) string {
+	entry := reflect.StructTag(tag).Get(matchTag)
+
+	// Protobuf tags are exciting, in that they /optionally/ have a json= section which takes
+	// precedence over the name= section.
+	if matchTag == "protobuf" {
```

Comment:
I think this is not needed for the first version. The protobuf stuff should be already covered via json. So we don't need this custom parsing for protobuf.

Reply:
The problem here is that the `json:` tag that is generated from tools is snake-cased, whereas it's possible to configure `UseProtoNames` to emit the camel-cased option in the protobuf: tag itself. Without this, I wouldn't be able to utilize tags. This is why I have the `json:` fallback to cover both use cases.

Reply:
Hmm understood, I dislike specific implementations that aren't general in goverter and also the internal precedence from json/name protobuf key.

What do you think about a more generic implementation where the user can configure the delimiters?
```
type TagMatch struct {
    Name      string
    Key       string
    FieldDelim string
    ValueDelim string
}

fieldDelim=, is the default

matchTag json
   is same as
matchTag json delimiter=,

matchTag protobuf valueDelim== key=name
matchTag protobuf valueDelim== key=json

matchTag gorm fieldDelm=; valueDelim=: key=column
https://gorm.io/docs/models.html
```